server/models/procedures/invoicing.sql

```diff
@@ -771,3 +771,79 @@ BEGIN
       HAVING balance <> 0
     ) AS zz ORDER BY zz.date;
 END $$
+
+
+-- This Procedure retrieve the balace(debit, credit) of each invoices during a specified period
+
+DROP PROCEDURE IF EXISTS `UnbalancedInvoicePayments`$$
+CREATE   PROCEDURE `UnbalancedInvoicePayments`(IN dateFrom DATE, IN dateTo DATE)
+BEGIN
+  DECLARE _uuid, _debtor_uuid, _tempRef BINARY(16);
+  DECLARE _invoiceID VARCHAR(100);
+  DECLARE _date DATE;
+
+  DECLARE done BOOLEAN;
+  DECLARE curs1 CURSOR FOR 
+    SELECT  i.uuid, i.debtor_uuid, i.date 
+    FROM invoice i
+    WHERE i.date BETWEEN dateFrom AND dateTo
+    ORDER BY DATE;
+   
+    DECLARE CONTINUE HANDLER FOR NOT FOUND SET done = TRUE;
+
+  -- all invoice balances will be stored temporarely in stage_invoice_balance table
+
+  CREATE TEMPORARY TABLE IF NOT EXISTS stage_invoice_balance(
+    tempRef BINARY(16),
+    invoice_uuid BINARY(16),
+    debit DECIMAL(19, 4),
+    credit DECIMAL(19, 4),
+    debtor_uuid BINARY(16),
+    DATE DATE,
+    reference VARCHAR(100)
+  );
+
+  SET _tempRef = HUID(UUID());
+
+  OPEN curs1;
+    read_loop: LOOP
+    FETCH curs1 INTO _uuid, _debtor_uuid, _date;
+      IF done THEN
+        LEAVE read_loop;
+      END IF;
+      -- some payment may be posted in the general ledger so it union  with posting_journal is required
+
+      SELECT IFNULL(d.text, '') INTO _invoiceID 
+      FROM document_map d WHERE d.uuid = _uuid
+      LIMIT 1;
+      -- check if the invoice is debited in stage_invoice_balance, this operation should not be done twise
+      -- the invoice must be debited once and after
+
+      INSERT INTO stage_invoice_balance
+      SELECT _tempRef, _uuid, SUM(invoices_payments.debit_equiv) AS debit, SUM(invoices_payments.credit_equiv) AS credit, _debtor_uuid, _date, _invoiceID
+      FROM
+	      ( SELECT  p.record_uuid AS UUID, p.debit_equiv, p.credit_equiv
+          FROM posting_journal p
+          WHERE p.record_uuid = _uuid AND p.entity_uuid = _debtor_uuid
+          UNION ALL
+          SELECT p.record_uuid AS UUID, p.debit_equiv, p.credit_equiv
+          FROM general_ledger p
+          WHERE  (p.record_uuid = _uuid AND p.entity_uuid = _debtor_uuid) OR (p.reference_uuid = _uuid AND p.entity_uuid = _debtor_uuid)
```

Comment:
In my experience, `OR` in SQL is super inefficient.   It would be better to break these into two different conditions and then put them together with `UNION ALL`.